### PR TITLE
chore: WIP remove trailing slash

### PR DESCRIPTION
**Pull request by ludder**

### CODE REVIEW

File: src/config/with-locale-router.ts [hunk 1]
```diff
@@ -2,15 +2,17 @@ import { NextConfig } from "next/dist/next-server/server/config"
...
 ): Partial<NextConfig> => {
+  console.log("trailingSlash: ", trailingSlash)
   return {
     ...nextConfig,
     i18n: config.toNextI18nConfig(),
     publicRuntimeConfig: {
       ...(nextConfig.publicRuntimeConfig ?? {}),
       [CONFIG_RUNTIME_KEY]: config.toObject(),
     },
+    trailingSlash,
```
Comment (LuudJanssen) [inline src/config/with-locale-router.ts]:
```suggestion
    trailingSlash: config.tralingSlash,
```

File: src/config/with-locale-router.ts [hunk 2]
```diff
@@ -2,15 +2,17 @@ import { NextConfig } from "next/dist/next-server/server/config"
 import { CONFIG_RUNTIME_KEY } from "../constants/config-runtime-key.constant"
 import config from "./index"
 
-export const withLocaleRouter = () => (
+export const withLocaleRouter = (trailingSlash = false) => (
```
Comment (LuudJanssen) [inline src/config/with-locale-router.ts]:
```suggestion
export const withLocaleRouter = () => (
```

File: src/strategy/strategy-investigator.class.ts [hunk 3]
```diff
@@ -27,6 +27,12 @@ export class StrategyInvestigator {
 
   private getStrategy(request: Request): ChainableStrategy {
     const url = getRequestUrl(request)
+    const hasTrailingSlash = url.href.endsWith("/")
+    const nextConfigTrailingSlash = this.config.trailingSlash
+
+    if (hasTrailingSlash && !nextConfigTrailingSlash) {
```
Comment (LuudJanssen) [inline src/strategy/strategy-investigator.class.ts]:
```suggestion
    if (hasTrailingSlash !== nextConfigTrailingSlash) {
```

File: src/strategy/strategy-investigator.class.ts [hunk 4]
```diff
@@ -27,6 +27,12 @@ export class StrategyInvestigator {
 
   private getStrategy(request: Request): ChainableStrategy {
     const url = getRequestUrl(request)
+    const hasTrailingSlash = url.href.endsWith("/")
```
Comment (LuudJanssen) [inline src/strategy/strategy-investigator.class.ts]:
```suggestion
    const hasTrailingSlash = url.pathname.endsWith("/")
```

`href` includes the query string, so you would miss calls to `/test/?query=true` with this approach.

**Review by LuudJanssen — CHANGES_REQUESTED** [review]

File: src/strategy/strategy-investigator.class.ts (hunk 4 above)
Comment (ludder) [reply src/strategy/strategy-investigator.class.ts]: I did this on purpose, but I guess you're right. Will change it.

Comment (ludder) [reply src/strategy/strategy-investigator.class.ts]: @LuudJanssen Please have another look

**Review by LuudJanssen — APPROVED** [review]

**Pull request merged**